src/common/utils/jsonp-helper.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import {Env} from 'playkit-js'
+
+/**
+ * jsonp callback function, returns the direct manifest uri
+ * @param {Object} data - the json object that returns from the server
+ * @param {string} uri - original request uri
+ * @returns {string} returns the direct uri
+ */
+function getDirectManfiestUri(data: Object, uri: string): string {
+  const getParsedUri = uri => {
+    const parser = document.createElement('a');
+    parser.href = uri;
+    return {
+      'hostname': parser.hostname,
+      'uri': uri
+    }
+  };
+
+  const uriHost = getParsedUri(uri).hostname;
+  const flavorUriHost = getParsedUri(data.flavors[0].url).hostname
+  if (data.flavors.length === 1 && uriHost !== flavorUriHost) {
+    return data.flavors[0].url;
+  } else {
+    return uri;
+  }
+}
+
+/**
+ * returns if we should use our jsonp http plugin
+ * @param {Object} config - configuration relevant to jsonp
+ * @returns {boolean} should or not use jsonp requests on manifests
+ */
+function shouldUseJsonp(config: Object): boolean{
+  if ((config && config.useJsonp) || Env.browser.name.includes("IE") || Env.browser.name.includes("Edge")){
```

Comment:
@odedhutzler we already know that Panasonic browser need it too - please add

Reply:
can we do better then browser sniffing?

Reply:
beautify

Reply:
@OrenMe I can change it to more configured... like
config.playback.options.adapters.forceRedriectBrowsers.includes(Env.browser.name) // "IE,Edge"